help_to_heat/frontdoor/interface.py

```diff
@@ -193,10 +226,38 @@ def get_session(self, session_id):
     @with_schema(load=CreateReferralSchema, dump=ReferralSchema)
     @register_event(models.Event, "Referral created")
     def create_referral(self, session_id):
-        session_data = api.session.get_session(session_id)
-        data = SupplierConverter(session_id).replace_in_session_data(session_data)
-        supplier = portal.models.Supplier.objects.get(name=data["supplier"])
-        referral = portal.models.Referral.objects.create(session_id=session_id, data=data, supplier=supplier)
+        answers = api.session.get_session(session_id)
+
+        # filter to only answers given on the user's journey
+        journey = calculate_journey(answers, confirm_and_submit_page)
+        given_answers = {}
+        for journey_page_name in journey:
+            page_answers = api.session.get_page_answers(session_id, journey_page_name)
+            given_answers = {**given_answers, **page_answers}
+
+        # override property type of park home
+        park_home_main_residence = given_answers.get(park_home_main_residence_field)
+        if park_home_main_residence == field_yes:
+            given_answers[property_type_field] = property_type_field_park_home
+            given_answers[property_subtype_field] = property_type_field_park_home
+
+        # override loft access if no loft
+        loft = given_answers.get(loft_field)
+        if loft == loft_field_no:
+            given_answers[loft_access_field] = loft_access_field_no_loft
+            given_answers[loft_insulation_field] = loft_insulation_field_no_loft
```

Comment:
I'm not quite sure what you mean by this; this code sets the answers for dependent answers that would be blank; it's mostly here to keep existing behaviour. the intent here was to move this logic to the time of submission so that it's not possible for these dummy answers to show on the change page

Reply:
Ok, I may have had it backwards then. If there's a comment we can leave to make clear why we're doing this ("setting the answers for dependent answers that would be blank" above) then that would be good.

Reply:
have added one now